build.cake

```diff
@@ -53,26 +53,29 @@ Task("Build")
 {
     var bepinExProperties = Directory("./BepInEx.Shared");
 
-    if(isBleedingEdge)
-    {
-        CopyFile(bepinExProperties + File("VersionInfo.cs"), bepinExProperties + File("VersionInfo.cs.bak"));
-        CopyFile(bepinExProperties + File("AssemblyInfo.cs"), bepinExProperties + File("AssemblyInfo.cs.bak"));
-        ReplaceRegexInFiles(bepinExProperties + File("VersionInfo.cs"), "([0-9]+\\.[0-9]+\\.[0-9]+\\.)[0-9]+", "${1}" + buildId);
+    buildVersion = FindRegexMatchGroupInFile(bepinExProperties + File("BepInEx.Shared.projitems"), @"\<Version\>([0-9]+\.[0-9]+\.[0-9]+)\<\/Version\>", 1, System.Text.RegularExpressions.RegexOptions.None).Value;
 
-        FileAppendText(bepinExProperties + File("AssemblyInfo.cs"), 
-            TransformText("\n[assembly: BuildInfo(\"BLEEDING EDGE Build #<%buildNumber%> from <%shortCommit%> at <%branchName%>\")]\n")
+    var buildSettings = new MSBuildSettings {
+        Configuration = "Release",
+        Restore = true
+    };
+
+    if (isBleedingEdge) 
+    {
+        buildSettings.Properties["BuildInfo"] = new[] {
+            TransformText("BLEEDING EDGE Build #<%buildNumber%> from <%shortCommit%> at <%branchName%>")
                 .WithToken("buildNumber", buildId)
                 .WithToken("shortCommit", currentCommit)
                 .WithToken("branchName", currentBranch)
-                .ToString());
-    }
+                .ToString()
+        };
 
-    buildVersion = FindRegexMatchInFile(bepinExProperties + File("VersionInfo.cs"), "[0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+", System.Text.RegularExpressions.RegexOptions.None);
+        buildSettings.Properties["AssemblyVersion"] = new[] { buildVersion + "." + buildId };
 
-    var buildSettings = new MSBuildSettings {
-        Configuration = "Release",
-        Restore = true
-    };
+        buildVersion += "-ci." + buildId;
```

Comment:
We call our CI builds "bleeding edge", so I think it's better to instead add `-be.` pre-release tag instead.